src/Evolution/EventsAndDenseTriggers/EventsAndDenseTriggers.hpp

```diff
@@ -234,6 +240,12 @@ void EventsAndDenseTriggers::run_events(
   ASSERT(heap_size_ != -1, "Not initialized");
   ASSERT(not events_and_triggers_.empty(),
          "Should not be calling run_events with no triggers");
+  using compute_tags = tmpl::remove_duplicates<tmpl::filter<
+      tmpl::flatten<tmpl::transform<
+          tmpl::at<typename Metavariables::factory_creation::factory_classes,
+                   Event>,
+          get_tags<tmpl::_1>>>,
+      db::is_compute_tag<tmpl::_1>>>;
```

Comment:
Should we just say all things in `compute_tags_for_observation_box` must be compute tags and then the filter is unnecessary?  (Again for plain EaT.)

Reply:
We could, but then I think each individual event needs to have a filter. The reason being that for example ObserveFields has as a template parameter all tags it wants to observe, both simple and compute. Requiring them to all be compute tags would mean we have to then check which tags are already in the box (evolved vars, etc.) and omit those. I guess each event could explicitly have as template parameters the simple tags, compute tags, and vars compute tags. The reason I didn't do this is because it's quite a bit more refactoring. We could require all the tags be compute tags and then just filter based on whether the simple (base) tag of the compute tag is retrievable. That still will require possibly writing compute tags for evolved variables, etc. or some other major refactor of individual events.

Reply:
OK.  This is fine, then.